### PR TITLE
SNOW-1692380: Release pipeline

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
# Overview

SNOW-1692380

## Pre-review self checklist
- [x] PR branch is updated with all the changes from `master` branch
- [x] The code is correctly formatted (run `mvn -P check-style validate`)
- [x] The pull request name is prefixed with `SNOW-XXXX: `
- [x] Code is in compliance with internal logging requirements

## External contributors - please answer these questions before submitting a pull request. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Issue: #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new dependency or upgrading an existing one

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -13,18 +13,22 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>8</maven.compiler.source>
-        <maven.compiler.target>8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
```
Comment (sfc-gh-dprzybysz) [inline pom.xml:17]: are we not going to support java 8? it's because of the hibernate 6.x?

File: public_pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (sfc-gh-dprzybysz) [inline public_pom.xml:1]: we are not shading anything, do we need to keep public pom as a separate file?

Comment (sfc-gh-pbulawa) [reply public_pom.xml:1]: I guess you are right, and we can merge these poms into a single one. I will explore it.

Comment (sfc-gh-pbulawa) [reply public_pom.xml]: Done, merged the poms.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -79,21 +80,21 @@ jobs:
...
-          { cloud: 'GCP', javaVersion: '11', testGroups: 'UNIT,STANDARD' },
-          { cloud: 'GCP', javaVersion: '17', testGroups: 'UNIT,STANDARD' },
-          { cloud: 'GCP', javaVersion: '21', testGroups: 'UNIT,STANDARD' },
-          { cloud: 'AZURE', javaVersion: '11', testGroups: 'UNIT,STANDARD' },
-          { cloud: 'AZURE', javaVersion: '17', testGroups: 'UNIT,STANDARD' },
-          { cloud: 'AZURE', javaVersion: '21', testGroups: 'UNIT,STANDARD' },
-          { cloud: 'AWS', javaVersion: '11', testGroups: 'UNIT,STANDARD,HYBRID' },
-          { cloud: 'AWS', javaVersion: '17', testGroups: 'UNIT,STANDARD,HYBRID' },
-          { cloud: 'AWS', javaVersion: '21', testGroups: 'UNIT,STANDARD,HYBRID' }
+          { cloud: 'AWS', javaVersion: '11', testGroups: 'UNIT,STANDARD,HYBRID', jdbcVersion: '3.13.31' },
```
Comment (sfc-gh-mkubik) [inline .github/workflows/build-test.yml:89]: what does HYBRID mean in this context and why is it only set for AWS?

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -79,21 +80,21 @@ jobs:
...
+          { cloud: 'AWS', javaVersion: '17', testGroups: 'UNIT,STANDARD,HYBRID', jdbcVersion: '3.15.1' },
+          { cloud: 'AWS', javaVersion: '21', testGroups: 'UNIT,STANDARD,HYBRID', jdbcVersion: 'LATEST' },
+          { cloud: 'AZURE', javaVersion: '11', testGroups: 'UNIT,STANDARD', jdbcVersion: 'LATEST' },
```
Comment (sfc-gh-mkubik) [inline .github/workflows/build-test.yml:92]: is there some mapping between jdbc and java version e.g. `3.13.31` -> java 11, `3.15.1`-> java 17, `LATEST`-> java 21? If yes then it's not kept here

File: .github/workflows/build-test.yml (hunk 3 above)
Comment (sfc-gh-pbulawa) [reply .github/workflows/build-test.yml:89]: It's hybrid tables that are in the PuPr for now - [docs](https://docs.snowflake.com/en/user-guide/tables-hybrid). It is supported only in AWS during this phase.

File: .github/workflows/build-test.yml (hunk 4 above)
Comment (sfc-gh-pbulawa) [reply .github/workflows/build-test.yml:92]: There's not. I wanted to test different jdbc versions with different Java versions (that's why these are mixed), but JDBC `3.13.31` does not support Java 21; thus, it is not present there.

**Review by sfc-gh-mkubik — APPROVED** [review]

**Pull request merged**